snakefile.py

```diff
@@ -246,3 +324,111 @@
 #     output: os.path.join(REPORT_DIR, '{sample}_kraken_report.html')
 #     log: os.path.join(LOG_DIR, 'kraken_report_{sample}.log')
 #     shell: "ktImportTaxonomy -m 3 -t 5 {input} -tax [taxonomy folder]??? -o {output} >> {log} 2>&1"
+
+
+# QC
+
+rule samtools_coverage: 
+    input: 
+        bed_file = AMPLICONS,
+        bam_file = os.path.join(MAPPED_READS_DIR, '{sample}_aligned.bam')
+    output: 
+        coverage = os.path.join(COVERAGE_DIR, '{sample}_coverage.csv'),
+        bedcov = os.path.join(COVERAGE_DIR, '{sample}_amplicons.csv')
+    log: os.path.join(LOG_DIR, 'coverage_{sample}.log')
+    # TODO add log
+    shell:
+        '''
+        samtools bedcov {input.bed_file} {input.bam_file} > {output.bedcov}
+        samtools coverage {input.bam_file} > {output.coverage}
+        '''
+        
+rule get_QC_table: 
+    input: 
+        coverage_file = os.path.join(COVERAGE_DIR, '{sample}_coverage.csv'),
+        amplicon_file = os.path.join(COVERAGE_DIR, '{sample}_amplicons.csv')
+    output: os.path.join(COVERAGE_DIR, '{sample}_merged_covs.csv')
+    log: os.path.join(LOG_DIR, 'coverage_parsing_{sample}.log')
+    # TODO add log
+    run:
+        def parsing_cov(coverage_file):
```

Comment:
Same comment as above: this is really familiar.  Looks like you copied code.  Please don't do that.  Just call the existing script.